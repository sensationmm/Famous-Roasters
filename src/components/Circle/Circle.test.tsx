import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { Circle, CircleType } from '.'

describe('Circle component', () => {
  it('Renders correctly', async () => {
    const { container } = render(<Circle value="sweet" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value experimentell & komplex', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="experimentell & komplex" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value fruchtig & lebhaft', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="fruchtig & lebhaft" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value floral & leicht', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="floral & leicht" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value nussig & schokoladig', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="nussig & schokoladig" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with value w체rzig & kr채ftig', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="w체rzig & kr채ftig" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for aroma tag with a different value', async () => {
    const { container } = render(<Circle type={CircleType.Aroma} value="something else" />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
