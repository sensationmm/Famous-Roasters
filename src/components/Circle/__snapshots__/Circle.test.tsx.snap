// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component Renders correctly 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-brand-grey-bombay"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with a different value 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-brand-grey-bombay"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with value experimentell & komplex 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-tags-capeHoney"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with value floral & leicht 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-tags-melrose"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with value fruchtig & lebhaft 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-tags-cottonCandy"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with value nussig & schokoladig 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-tags-coldTurkey"
  />
</div>
`;

exports[`Circle component Renders correctly for aroma tag with value würzig & kräftig 1`] = `
<div>
  <div
    class="inline-flex rounded-full w-6 h-6 md:w-8 md:h-8 xl:w-12 xl:h-12 bg-tags-yourPink"
  />
</div>
`;
